qunicorn_core/core/pilotmanager/qmware_pilot.py

```diff
@@ -109,41 +109,38 @@ def run(
         results = []
 
         for qmware_job_id, program in zip(qmware_job_ids, programs):
-            results.extend(QMwarePilot._get_job_results(qunicorn_job, qmware_job_id, program))
+            try:
+                results.extend(QMwarePilot._get_job_results(qunicorn_job, qmware_job_id, program))
+            except RetryError:
+                error = QunicornError("QMware job timed out")
+                qunicorn_job.save_error(error)
+                raise error
 
         return results, JobState.FINISHED
 
     @staticmethod
+    @retry(
+        retry=retry_if_result(lambda output: output is None),
+        stop=stop_after_attempt(30),
```

Comment:
How long will this keep polling (maybe add a comment with a rough estimate, or use a time based stop?)